source/mind_ir.js

```diff
@@ -0,0 +1,755 @@
+            if (!(arg.name.includes("MakeTuple") || arg.name.includes("UpdateState") || arg.name.includes("Load") || arg.name.includes("Constant"))) {
+                this._outputs.push(new mind_ir.Parameter(j.toString(), arg ? [arg] : []));
+            }
+        }
+
+        for (const attr of node.attribute) {
+            //debugger;
+            if (!attr.name.includes("input_names") && !attr.name.includes("output_names")) {
+                this._attributes.push(new mind_ir.Attribute(this, attr));
+            }
+        }
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get type() {
+        return this._type;
+    }
+
+    get inputs() {
+        return this._inputs;
+    }
+
+    get outputs() {
+        return this._outputs;
+    }
+
+    get attributes() {
+        return this._attributes;
+    }
+};
+
+mind_ir.Utility = class {
+
+    static convertToArray(values) {
+        return values.map((value) => {
+            for (const array of Object.values(value)) {
+                if (Array.isArray(array) && array.length > 0) {
+                    return array[0];
+                }
+            }
+            return null;
+        });
+    }
+
+    static convertList(lst) {
+        const int32Array = new Int32Array(lst.length);
+
+        lst.forEach((int64, index) => {
+            int32Array[index] = int64.low;
+        });
+        return int32Array;
+    }
+
+    static enum(name, value) {
+        const type = name && mind_ir.schema ? mind_ir.schema[name] : undefined;
+        if (type) {
+            mind_ir.Utility._enumKeyMap = mind_ir.Utility._enumKeyMap || new Map();
+            if (!mind_ir.Utility._enumKeyMap.has(name)) {
+                const map = new Map();
+                for (const key of Object.keys(type)) {
+                    map.set(type[key], key);
+                }
+                mind_ir.Utility._enumKeyMap.set(name, map);
+            }
+            const map = mind_ir.Utility._enumKeyMap.get(name);
+            if (map.has(value)) {
+                return map.get(value);
+            }
+        }
+        return value;
+    }
+};
+
+mind_ir.AttributeType = {
+    UNDEFINED: 0,
+    FLOAT: 1,
+    UINT8: 2,
+    INT8: 3,
+    UINT16: 4,
+    INT16: 5,
+    INT32: 6,
+    INT64: 7,
+    STRING: 8,
+    BOOL: 9,
+    FLOAT16: 10,
+    DOUBLE: 11,
+    UINT32: 12,
+    UINT64: 13,
+    COMPLEX64: 14,
+    COMPLEX128: 15,
+    BFLOAT16: 16,
+    TENSOR: 17,
+    GRAPH: 18,
+    TENSORS: 19,
+    TUPLE: 20,
+    LIST: 21,
+    DICT: 22,
+    UMONAD: 23,
+    IOMONAD: 24,
+    NONE: 25,
+    PRIMITIVECLOSURE: 26,
+    FUNCGRAPHCLOSURE: 27,
+    PARTIALCLOSURE: 28,
+    UNIONFUNCCLOSURE: 29,
+    CSR_TENSOR: 30,
+    COO_TENSOR: 31,
+    ROW_TENSOR: 32,
+    CLASS_TYPE: 33,
+    NAME_SPACE: 34,
+    SYMBOL: 35,
+    TYPE_NULL: 36,
+    MAP_TENSOR: 37
+};
+
+mind_ir.Attribute = class {
+
+    constructor(context, attribute) {
+
+        this._name = attribute.name;
```

Comment:
`Attribute` has been removed in latest and replaced with using `Argument` to represent attribtues.